pallets/msa/src/lib.rs

```diff
@@ -614,9 +620,10 @@ pub mod pallet {
 			Self::delete_key_for_msa(msa_id, &who)?;
 			Self::deposit_event(Event::KeyRemoved { key: who });
 
-			// Deposit the "MsaRetired" event
 			Self::deposit_event(Event::MsaRetired { msa_id });
-			Ok(())
+
+			// MSA retirement is free
+			Ok(Pays::No.into())
```

Comment:
you can remove this

Reply:
Ahh. Perhaps I am confused.

In this case, we are charging a fee when it errors. I am not sure if this is intentional.

Reply:
Yes, it's conditional.  It's free for valid requests but we charge a fee for invalid requests.

Reply:
I am not opposed to this idea. However, we do not do this with any other free transaction. Can you please share with me why we decided to do it this way for retiring?

Reply:
I have redesigned it to use the signed extension.